src/xdocs/writingchecks.xml

```diff
@@ -423,34 +420,36 @@ public class MethodLimitCheck extends AbstractCheck
 
       <p>
         Before <a href="releasenotes_old_6-0_7-8.html#Release_6.0">Checkstyle 6.0</a>,
-        there was no comments in AST tree as comments are not a code and ignored by compiler,
-        so it was not a primary focus of Checkstyle.
-        But more and more requests appear to do validation of code vs javadoc or comments.
-        So there was a simple solution to receive plain text of file in Check, do own parsing and
-        searching for required comments in file.
+        there were no comments in the AST. Comments are not a code and are ignored by compiler,
+        so they were not a primary focus of Checkstyle.
+        But more and more requests appeared to do validation of javadoc or comments.
+        One simple solution was to receive the plain-text file in a Check, manually parse it and
+        validate the comments in the file.
       </p>
       <p>
-        Since 6.0, there is a new method in AbstractCheck class that allow you to see or not
-        comment nodes in AST Tree -
+        Since version 6.0, there is a new method in the AbstractCheck class that allows you to see
+        comment nodes in the AST -
         <a href="apidocs/com/puppycrawl/tools/checkstyle/api/AbstractCheck.html#isCommentNodesRequired--">
-        isCommentNodesRequired()</a>. It should return TRUE if Check want to see comments in AST
-        Tree.
+        isCommentNodesRequired()</a>. When this returns `true`, comments are included in the AST.
       </p>
       <p>
-        It is done as optional because it is change for a AST Tree structure, and non of existing
-        Checks ready for this.
-        Checkstyle does not do re-parse file one more time, comments were already in grammar and
-        parsed, but skipped during AST nodes creation.
+        This is optional to maintain backwards compatibility: it changes for the AST structure,
+        and breaking existing checks.
+        Checkstyle does not parse the files twice: comments were already in the grammar and
+        parsed, but skipped during the creation of AST nodes.
```

Comment:
`Checkstyle does not parse the files twice:` => `Checkstyle does not parse the files twice to bring forward the comments.`
Just capitalize next sentence.

Reply:
changed to `Checkstyle does not parse the files twice to include the comments. Comments were already in the grammar and parsed, but skipped during the creation of AST nodes.`